src/SpeedyTransforms/spectral_transform.jl

```diff
@@ -513,12 +509,28 @@ function gridded(   alms::AbstractMatrix{T};            # spectral coefficients
     return gridded(alms, S; kwargs...)
 end
 
+"""
+$(TYPEDSIGNATURES)
+Spectral transform (spectral to grid space) from spherical coefficients `alms` to a newly allocated gridded
+field `map`. Based on the size of `alms` the grid type `grid`, the spatial resolution is retrieved based
+on the truncation defined for `grid`. SpectralTransform struct `S` is allocated to execute `gridded(alms, S)`."""
+function gridded(   alms::LowerTriangularMatrix{T};            # spectral coefficients
+                    recompute_legendre::Bool = true,    # saves memory
+                    Grid::Type{<:AbstractGrid} = DEFAULT_GRID,
+                    kwargs...
+                    ) where {NF, T<:Complex{NF}}        # number format NF
+
+    lmax, mmax = matrix_size(alms) .- 1                        # -1 for 0-based degree l, order m
+    S = SpectralTransform(NF, Grid, lmax, mmax; recompute_legendre)
+    return gridded(alms, S; kwargs...)
+end
+
 """
 $(TYPEDSIGNATURES)
 Spectral transform (spectral to grid space) from spherical coefficients `alms` to a newly allocated gridded
 field `map` with precalculated properties based on the SpectralTransform struct `S`. `alms` is converted to
 a `LowerTriangularMatrix` to execute the in-place `gridded!`."""
-function gridded(   alms::AbstractMatrix,       # spectral coefficients
+function gridded(   alms::Union{AbstractMatrix, LowerTriangularMatrix},       # spectral coefficients
```

Comment:
I think I want to disable the use of `gridded(M::Matrix)` and `spectral(M::Matrix)`, people should be explicit that their data represents spectral/gridded data and do `gridded(LowerTriangularMatrix(M))` or `spectral(FullGaussianGrid(M))` manually...